src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
 {
-    uint256 hash;
-    hash.SetHex (amendmentID);
-
     if (hash.isZero ())
     {
         assert (false);
-        return nullptr;
+        return false;
     }
 
-    AmendmentState* f = getCreate (hash, true);
+    ScopedLockType sl (mLock);
+    AmendmentState* s = getCreate (hash, true);
```

Comment:
I don't care for naked pointers in general, but I think that a naked pointer is fine here - the code here doesn't need to participate in object lifetime decisions so doesn't need `unique_ptr` or `shared_ptr` and the lifetime of the object being pointed to is guaranteed by external code. Additionally, the code in question is a pure implementation detail and the pointers don't escape. 

I guess this could, in theory, be a reference (and throwing from `getCreate` in the unlikely event that creation failed) but I wouldn't worry too much about this.